app/javascript/components/set-ownership-form/index.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Grid } from 'patternfly-react';
+import MiqFormRenderer from '../../forms/data-driven-form';
+import { http, API } from '../../http_api';
+import createSchema from './ownership-form.schema';
+
+class SetOwnershipForm extends Component {
+  state = {
+    initialValues: {},
+    userOptions: [],
+    groupOptions: [],
+  };
+
+  componentDidMount() {
+    const { ownershipIds } = this.props;
+    this.loadInitialData(ownershipIds);
+  }
+
+  checkIfTenantGroup = groupId => new Promise((resolve, reject) => {
+    API.get(`/api/tenant_groups/${groupId}`, { skipErrors: true })
```

Comment:
I don't like the idea of the async here, all the requests should be sent from a single place, it's safer. Can you chain this after the `Promise.all`?